pkg/generate/app/app.go

```diff
@@ -154,7 +154,9 @@ func (s *BuildStrategyRef) BuildStrategy() (*buildapi.BuildStrategy, []buildapi.
 	return &buildapi.BuildStrategy{
 		Type: buildapi.STIBuildStrategyType,
 		STIStrategy: &buildapi.STIBuildStrategy{
-			Image: s.Base.NameReference(),
+			From: kapi.ObjectReference{
```

Comment:
the generator can only point to an ImageRepo reference if either
1) it's going to create the ImageRepo
or
2) you know the ImageRepo exists